htdocs_symfony/src/Controller/Backend/CachesController.php

```diff
@@ -0,0 +1,116 @@
+        $fetched_caches = '0';
+
+        // create input field for caches_by_searchfield
+        $form = $this->createForm(CachesFormType::class);
+
+        // see: https://symfonycasts.com/screencast/symfony-forms/form-submit
+        // handles the request (submit-button of the form), but only if there is a POST request
+        $form->handleRequest($request);
+        // if is true only if there is a request submitted and it is valid
+        if ($form->isSubmitted() && $form->isValid()) {
+            // read content of form input field
+            $input_data = $form->getData();
+
+            // send request to DB
+            $fetched_caches = $this->search_by_searchfield($connection, $input_data["content_caches_searchfield"]);
+
+//            dd($input_data["content_caches_searchfield"]);
+//            dd($fetched_caches);
+//            die();
+        }
+
+        // load all caches from database and hand over to twig page
+        // just for initial test to learn how it works.. leave it here for later check up
+        // $caches = $connection->fetchAll('SELECT * FROM caches');
+
+        return $this->render('backend/caches/index.html.twig', ['cachesForm' => $form->createView(), 'caches_by_searchfield' => $fetched_caches]);
+    }
+
+    /**
+     * @Route("/caches/caches_by_searchfield", name="create_form_caches_by_searchfield")
+     */
+    public function create_form_caches_by_searchfield(EntityManagerInterface $em)
+    {
+        $form = $this->createForm(CachesFormType::class);
+
+        return $this->render('backend/caches/index.html.twig', ['cachesForm' => $form->createView()]);
+    }
+
+    /**
+     * @Route("/cache/{wp_oc}", name="cache_by_wp_oc")
+     */
+    public function search_by_cache_id(Connection $connection, string $wp_oc): Response
+    {
+        $wp_oc = strtoupper($wp_oc);
+
+        // search in database for wp_oc with the given {wp_oc}
+        $fetched_cache = $connection->fetchAll('SELECT * FROM caches WHERE wp_oc = "' . $wp_oc . '"');
+
+        // if search for wp_oc gave no result, search for wp_gc
+        if (empty($fetched_cache) == true) {
+            $fetched_cache = $connection->fetchAll('SELECT * FROM caches WHERE wp_gc = "' . $wp_oc . '"');
+
+            // if search for wp_gc also gave no result, search for wp_nc
+            if (empty($fetched_cache) == true)
+                $fetched_cache = $connection->fetchAll('SELECT * FROM caches WHERE wp_nc = "' . $wp_oc . '"');
+        }
+
+//        if (!$fetched_cache)
+//            print_r("nope, nix da");
+//        else
+//            print_r($fetched_cache);
+//        die();
+
+        return $this->render('backend/caches/index.html.twig', ['cache_by_id' => $fetched_cache]);
+    }
+
+//    /**
+//     * @Route("/caches/list", name="caches_by_searchfield", methods="POST")
+//     */
+    function search_by_searchfield(Connection $connection, string $searchtext)
+    {
+
+        if ($searchtext != "") {
```

Comment:
Ask @nlubisch for help - Insecure SQL injection possible.

Reply:
ok, waiting for advice how to avoid insecure SQL injection